### PR TITLE
A simple webpage to present HighFive for PASC22

### PR DESCRIPTION
**Pull request by ohm314**
The poster contains a big QR code for people to scan. Instead of just sending them to the github page, I thought it would be nice to have a simple page that presents library features with examples. Basically all the examples we wouldn't want to have on the poster itself.

## What's missing
- [x] More text explaining the features. It doesn't have to be a lot, but this is maybe a bit minimalistic.
- [ ] An example on slices, hyperslabs, more complex data arrangement?
- [ ] An example on XTensor?
- [ ] An example and some intro information on H5Easy?
- [x] Links to the documentation and installation instructions.

## How to view the page
Since I coded this up with some javascript that loads the C++ examples the page cannot be viewed locally and one has to view it via a HTTP server. After checking out the branch, navigate to `doc/poster` and run `python3 -m http.server 8888`.

### CODE REVIEW

File: doc/poster/example1_hdf5.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  // writing the data
+  status = H5Dwrite(dset_id, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, 
+      data.data());
+
+  // === Reading back from HDF5 into a vector
+
+  // Retrieve result size and preallocate vector
+  std::vector<double> result;
+  dspace_id = H5Dget_space(dset_id);
```
Comment (matz-e) [inline doc/poster/example1_hdf5.cpp:36]:
```suggestion
  dset_id = H5Dopen(file_id, "/group/dset1", H5P_DEFAULT);
  dspace_id = H5Dget_space(dset_id);
```

File: doc/poster/example1_hdf5.cpp [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      H5P_DEFAULT);
+
+  // creating a dataset
+  dset_id = H5Screate_simple(1, dims, NULL);
+  dset_id = H5Dcreate2(group_id, "group/dset1", H5T_STD_I32BE, dspace_id,
+      H5P_DEFAULT, H5P_DEFAULT, H5P_DEFAULT);
+
+  // writing the data
+  status = H5Dwrite(dset_id, H5T_NATIVE_INT, H5S_ALL, H5S_ALL, H5P_DEFAULT, 
+      data.data());
```
Comment (matz-e) [inline doc/poster/example1_hdf5.cpp:30]:
```suggestion
      data.data());

  // close dataset after writing
  status = H5Dclose(dset_id);
```

File: doc/poster/example1_hdf5.cpp (hunk 1 above)
Comment (matz-e) [reply doc/poster/example1_hdf5.cpp:36]: Then we're a bit closer to the HighFive example.

File: doc/poster/example2.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#include <HighFive>
+
+using HighFive;
+
+void write_data() {
+  FileDriver fdrv;
```
Comment (matz-e) [inline doc/poster/example2.cpp:6]: I prefer to name my Drivers Adam ;)

File: doc/poster/pasc22.html [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                <pre class="font-monospace"><div class="example" id="example5"></div></pre>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+
+    </div>
+    <div class="container">
+      <p>HighFive is both thread-safe and supports the HDF5 MPI backend and can be natively used in
```
Comment (matz-e) [inline doc/poster/pasc22.html:150]:
```suggestion
      <p>HighFive supports the HDF5 MPI backend and can be natively used in
```
Thread-safety is a feature of HDF5 IMO…

**Review by matz-e — COMMENTED** [review]
First pass; I'm still trying to get the pure HDF5 example to work locally

File: doc/poster/example1_hdf5.cpp [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  // Opening a file
+  file_id = H5Fopen("/tmp/new_file.h5", H5F_ACC_RDWR, H5P_DEFAULT);
+
+  // Creating a group
+  group_id = H5Gcreate2(file_id, "/group", H5P_DEFAULT, H5P_DEFAULT,
+      H5P_DEFAULT);
+
+  // creating a dataset
+  dset_id = H5Screate_simple(1, dims, NULL);
+  dset_id = H5Dcreate2(group_id, "group/dset1", H5T_STD_I32BE, dspace_id,
```
Comment (matz-e) [inline doc/poster/example1_hdf5.cpp:25]:
```suggestion
  dspace_id = H5Screate_simple(1, dims, NULL);
  dset_id = H5Dcreate2(group_id, "dset1", H5T_STD_I32BE, dspace_id,
```

File: doc/poster/example1_hdf5.cpp [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  // ==== Writing a simple vector to HDF5
+
+  // Setting up dataset dimensions and input data
+  int ndims = 1;
+  hsize_t dims[ndims];
+  dims[0]      = 50;
+  std::vector<double> data(50, 1);
+
+  // Opening a file
+  file_id = H5Fopen("/tmp/new_file.h5", H5F_ACC_RDWR, H5P_DEFAULT);
```
Comment (matz-e) [inline doc/poster/example1_hdf5.cpp:17]:
```suggestion
  file_id = H5Fcreate("/tmp/new_file.h5", H5F_ACC_TRUNC, H5P_DEFAULT);
```
(or don't call it "new" file :))

**Review by matz-e — COMMENTED** [review]
Round of fixes after compiling the code and running it.

File: doc/poster/example2.cpp (hunk 3 above)
Comment (ohm314) [reply doc/poster/example2.cpp:6]: too biblical :P

File: doc/poster/pasc22.html (hunk 4 above)
Comment (ohm314) [reply doc/poster/pasc22.html:150]: I had a doubt there myself, but had taken it from the abstract.

**Comment by ohm314** [comment]
Hi @tdegeus , I think this page would be a great place to also briefly present H5Easy, if you wish. If you want, could you share with me one paragraph of text and a couple of short code snippets? I would get them all in the right format and present them towards the end in a small section with a tabbed box, like I do for the supported datatypes.

**Comment by tdegeus** [comment]
That would indeed be great @ohm314 . Is it ok if I get to it somewhere this week ?

**Comment by ohm314** [comment]
No problem - I think we can be relaxed about this page and add things also when they are ready. If you have it by end of week that would be really great though.

**Review by matz-e — APPROVED** [review]
Good to merge!

(as a future todo: on my screen, the Godbolt logo is really tiny, maybe we could make it larger, _but only after merging_)

**Comment by ohm314** [comment]
I'm going to merge this - not because we're done, but rather because I want to see if we get a proper webpage live. If fixes are needed (and for all improvements) there will be follow-up PR

**Pull request merged**